packages/web-components/fast-components/src/color/recipes/accent-foreground.ts

```diff
@@ -1,19 +1,19 @@
-import { PaletteRGB } from "../palette";
-import { Swatch, SwatchRGB } from "../swatch";
+import { Palette } from "../palette";
+import { Swatch } from "../swatch";
 import { directionByIsDark } from "../utilities/direction-by-is-dark";
 
 /**
  * @internal
  */
 export function accentForeground(
-    palette: PaletteRGB,
+    palette: Palette,
     reference: Swatch,
     contrastTarget: number,
     restDelta: number,
     hoverDelta: number,
     activeDelta: number,
     focusDelta: number
-): Record<"rest" | "hover" | "active" | "focus", SwatchRGB> {
+): Record<"rest" | "hover" | "active" | "focus", Swatch> {
```

Comment:
I'm curious why you're converting all of these to `Swatch` from `SwatchRGB`; I thought we needed RGB info to convert these to alpha colors eventually?

Reply:
Per our chat, my initial goal here was to simplify the interface to avoid restrictions to future consumers. Potentially we could support other color models or something.
I had forgotten about making this change in the context of transparency support, but I wasn't specific enough in our chat long ago about needing the RGB info. The underlying issue with the previous system was that we passed around solid hex values everywhere. We could parse those and use them in child component recipes, but as soon as a container needed to use a transparent color, there was no longer a way to parse that or use it to create derivative colors.
Now that we have the more descriptive palette and swatch model, and we're getting an actual object any time we evaluate a recipe, I believe there's a small augmentation we can add to Swatch to allow transparent colors to be rendered in css, but passed around with sufficient information to make derivative colors.
We agreed to leave this change in, as even if we end up needing to diverge to fully support transparency, the addition of RGB information alone is not sufficient (at least in my current assumed model) to do what we would need to,